semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java

```diff
@@ -163,7 +165,39 @@ public static boolean isSqlTypeBackendForOtelFormat(
     if (SpanAttributeUtils.containsAttributeKey(event, SQL_URL)) {
       return Optional.of(SpanAttributeUtils.getStringAttribute(event, SQL_URL));
     }
-    return getBackendURIForOtelFormat(event);
+    Optional<String> backendUrl = getBackendURIForOtelFormat(event);
+    if (backendUrl.isPresent()) {
+      return backendUrl;
+    }
+    if (SpanAttributeUtils.containsAttributeKey(
+        event, OTelDbSemanticConventions.DB_CONNECTION_STRING.getValue())) {
+      String url = SpanAttributeUtils.getStringAttribute(
+          event, OTelDbSemanticConventions.DB_CONNECTION_STRING.getValue());
+      if (!isValidURI(url)) {
```

Comment:
nit, if you do `isValirURI(url) { return Optional.of(url); }` I think you can just keep the last `return Optional.empty()`.

Reply:
Sure, I will do it in a subsequent pr.